rust_src/build.rs

```diff
@@ -62,23 +126,24 @@ fn export_function(
     name: Option<String>,
     line: &str,
     modname: &str,
-    lineno: u32,
+    lineno: i32,
     msg: &str,
-) -> Option<String> {
+) -> Result<Option<String>, LintMsg> {
```

Comment:
Can we consolidate this into just a Result<String, LintMsg> instead of a Result<Option<String, ...> ? It feels unidiomatic to me in this case.

Reply:
/home/shaleh/repos/opensource/rust/src $ grep -r -- 'Result<Option' . | wc -l
70

Reply:
This is what a strong type system gives you. Falling back to an empty string is weaker and carries less semantic meaning.

Reply:
That is a good point, and I agree that there are many cases where Result<Option<T>, U> carries important semantic meaning. My question was more aimed at our specific usage. We use the result of export_function in two places, in one case, if our option is in the None case, we just panic anyway, implying an error condition. In the other case, we do nothing, which backs your point of the option having semantic meaning.
 
My personal philosophy is "aggressive minimalism", so I like to always question usages of type pattens like this, because I feel they can be intimidating to newer potential contributors. However, after thinking about it some more, I think the code is fine as is.

tdlr; question withdrawn 😃

Reply:
Please keep questioning. I approve of minimalism.

As for newbies, clear code following idioms is how we learn. Otherwise it is C with window dressing